src/deluge/model/note/note_row.cpp

```diff
@@ -194,6 +194,167 @@ Error NoteRow::beenCloned(ModelStackWithNoteRow* modelStack, bool shouldFlattenR
 	return error;
 }
 
+// initialize row square info array before we iterate through the note row to store square info
+void NoteRow::initRowSquareInfo(SquareInfo rowSquareInfo[kDisplayWidth], bool anyNotes) {
+	for (int32_t x = 0; x < kDisplayWidth; x++) {
+		initSquareInfo(rowSquareInfo[x], anyNotes, x);
+	}
+}
+
+void NoteRow::initSquareInfo(SquareInfo& squareInfo, bool anyNotes, int32_t x) {
```

Comment:
this should be a constructor on SquareInfo that takes anyNotes and x as arguments